src/misc/programs.cpp

```diff
@@ -97,12 +97,10 @@ static Bitu PROGRAMS_Handler(void) {
 	                         256 + static_cast<uint16_t>(exec_block_size));
 	HostPt writer=(HostPt)&index;
 	for (;size>0;size--) *writer++=mem_readb(reader++);
-	Program * new_program;
 	if (index >= internal_progs.size()) E_Exit("something is messing with the memory");
-	PROGRAMS_Main * handler = internal_progs[index];
-	(*handler)(&new_program);
+	PROGRAMS_Creator creator = internal_progs[index];
+	auto new_program = creator();
```

Comment:
The `unique_ptr`'s content isn't being replaced/exchanged, so can become `const auto` (for it's very brief lifetime :fly: )